### PR TITLE
Better Upload exceptions in Uploader.php

### PR DESCRIPTION
**Pull request by sicet7**
### Description (*)
This pull request simply adds more exceptions messages to the uploader.
The exception messages is copied from: https://www.php.net/manual/en/features.file-upload.errors.php#115746

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/Varien/File/Uploader.php [hunk 1]
```diff
@@ -155,6 +168,10 @@ function __construct($fileId)
     {
         $this->_setUploadFileId($fileId);
         if(!file_exists($this->_file['tmp_name'])) {
+            $errorCode = (isset($this->_file['error']) ? $this->_file['error'] : false);
+            if ($errorCode !== false && in_array($errorCode, self::UPLOAD_ERRORS)) {
```
Comment (kiatng) [inline lib/Varien/File/Uploader.php:172]: To bypass the check on `in_array` in cases there is no error, consider this:

```php
$errorCode = $this->_file['error'] ?? 0;
if ($errorCode && in_array($errorCode, self::UPLOAD_ERRORS)) {
```

**Review by tobihille — APPROVED** [review]
Better than my core hack :-D

File: lib/Varien/File/Uploader.php (hunk 1 above)
Comment (fballiano) [inline lib/Varien/File/Uploader.php:172]:
```suggestion
            $errorCode = $this->_file['error'] ?? 0;
            if ($errorCode && in_array($errorCode, self::UPLOAD_ERRORS)) {
```

Comment (fballiano) [reply lib/Varien/File/Uploader.php:172]: commiting @kiatng's code to try to move this forward

**Comment by fballiano** [comment]
actually, I can't commit cause this PR is not open for external contributions. I'll mark is as draft. @sicet7 could you commit the suggestion or open the PR for contributions?

**Comment by sicet7** [comment]
@fballiano i am sadly not able to modify anything on the fork, as i am no longer part of the organization who is in control of the fork, for that reason i am simply going to close this pull request.

**Comment by ADDISON74** [comment]
@fballiano - could you please take care of this PR?

**Comment by fballiano** [comment]
> @fballiano - could you please take care of this PR?

Absolutely!

**Pull request closed without merging**